src/themes/components/modal.ts

```diff
@@ -0,0 +1,102 @@
+import { mode } from '@chakra-ui/theme-tools';
+import {
+  createMultiStyleConfigHelpers,
+  defineStyle,
+} from '@chakra-ui/styled-system';
+import { modalAnatomy as parts } from '@chakra-ui/anatomy';
+
+import { opacity } from '../shared/opacity';
+
+const { defineMultiStyleConfig, definePartsStyle } =
+  createMultiStyleConfigHelpers(parts.keys);
+
+const baseStyleOverlay = defineStyle({
+  bg: 'black',
+  zIndex: 'modal',
+  opacity: `${opacity[40]} !important`,
+});
+
+const baseStyleDialogContainer = defineStyle({
+  display: 'flex',
+  zIndex: 'modal',
+  justifyContent: 'center',
+  alignItems: 'center',
+  overflow: 'hidden',
+  overscrollBehaviorY: 'none',
+});
+
+const baseStyleDialog = defineStyle({
+  borderRadius: 'sm',
+  bg: mode('white', 'gray.700'),
+  color: 'inherit',
+  my: '16',
+  zIndex: 'modal',
+  boxShadow: mode('lg', 'dark-lg'),
+});
+
+const baseStyleHeader = defineStyle({
+  px: 'lg',
+  py: 'xl',
+  fontSize: 'md',
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  fontWeight: 'semibold',
+  position: 'relative',
+});
+
+const baseStyleCloseButton = defineStyle({
+  width: '18px',
+  height: '18px',
+  fontSize: '18px',
+});
+
+const baseStyleBody = defineStyle({
+  p: '2xl',
+  flex: '1',
+});
+
+const baseStyleFooter = defineStyle({
+  px: 'md',
+  py: 'lg',
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+});
+
+const baseStyle = definePartsStyle({
+  overlay: baseStyleOverlay,
+  dialogContainer: baseStyleDialogContainer,
+  dialog: baseStyleDialog,
+  header: baseStyleHeader,
+  closeButton: baseStyleCloseButton,
+  body: baseStyleBody,
+  footer: baseStyleFooter,
+});
+
+function getSize(value: string) {
```

Comment:
if you pass `size=full`, nothing happens. Could you please check? Maybe we don't need it at all though